src/cloud-cli/tests/api/auth/test_callback_server.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+import pytest
+from flask import Flask
+from flask.testing import FlaskClient
+from werkzeug.test import TestResponse
+
+from meltano.cloud.api.auth import callback_server
+from meltano.cloud.api.config import MeltanoCloudConfig
+
+
+class TestMeltanoCloudAuthCallbackServer:
+    """Test the Meltano Cloud Auth Callback Server."""
+
+    @pytest.fixture(scope="function")
+    def config(self, tmp_path: Path):
+        return MeltanoCloudConfig(config_path=tmp_path / "meltano-cloud.json")
+
+    @pytest.fixture(scope="function")
+    def subject(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        tmp_path: Path,
+        config: MeltanoCloudConfig,
+    ):
+        monkeypatch.setenv("MELTANO_CLOUD_CONFIG_PATH", str(config.config_path))
+        return callback_server.APP
+
+    @pytest.fixture
+    def client(self, subject: Flask):
+        return subject.test_client()
+
+    def test_callback_page(self, client: FlaskClient):
+        response = client.get("/")
+        assert (
+            b'<p id="start-message">Logging in to Meltano Cloud...</p>' in response.data
+        )
+
+    def test_handle_tokens(
+        self, subject: Flask, client: FlaskClient, config: MeltanoCloudConfig
+    ):
+        response: TestResponse = client.get(
+            "/tokens?id_token=meltano-cloud-testing&access_token=meltano-cloud-testing"
+        )
+        config.refresh()
+        assert response.status_code == 204
+        assert config.access_token == "meltano-cloud-testing"  # noqa: S105
+        assert config.id_token == "meltano-cloud-testing"  # noqa: S105
+        with open(config.config_path) as config_file:
```

Comment:
```suggestion
        with Path(config.config_path).open() as config_file:
```

Reply:
@edgarrmondragon Why prefer `Path.open` over `open`?

Reply:
Really just a nit but I like the API better, and I guess I've been looking at Ruff-linted code 🙈

Reply:
I see: https://beta.ruff.rs/docs/rules/#flake8-use-pathlib-pth

> `open()` should be replaced by `Path.open()`

I'm not a huge fan of that rule in particular (`PTH123`), but Ruff can automatically make it consistent, and I value consistency, so I'll support it in most cases.

Unfortunately it appears that `Path.open` doesn't support all of the parameters that `open` does. Notably it's missing the `opener` parameter, which is useful for opening a file with non-default permission bits set. We open the Cloud CLI config/secrets file for writing with 0600 as its permissions using that.